src/Umbraco.Web/Editors/MediaController.cs

```diff
@@ -278,6 +281,31 @@ public IEnumerable<MediaItemDisplay> GetByIds([FromUri]int[] ids)
         }
 
         /// <summary>
+        /// Searches media and returns a paged result
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="pageNumber"></param>
+        /// <param name="pageSize"></param>
+        /// <param name="searchFrom"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [FilterAllowedOutgoingMedia(typeof(IEnumerable<ContentItemBasic<ContentPropertyBasic, IMedia>>), "Items")]

```

Comment:
@madsrasmussen We also already have the GetChildren method which supports searching for the entire tree if you passed in "-1" as the Id, this uses a SQL search which is what we want, I don't want Examine searches to be used for core pickers, this should be done with the source data which is SQL. Is there a reason why GetChildren was not considered?